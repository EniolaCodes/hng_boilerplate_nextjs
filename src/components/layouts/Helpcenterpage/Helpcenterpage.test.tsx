import { fireEvent, render, screen } from "@testing-library/react";

import "@testing-library/jest-dom";

import HelpCenterPage from "../../../app/help-center/page";

describe("helpCenterPage", () => {
  it("renders Help Center title", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const titleElement = screen.getByText("Help Center");
    expect(titleElement).toBeInTheDocument();
  });

  it("renders the main heading", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const headingElement = screen.getByText("How can we help You?");
    expect(headingElement).toBeInTheDocument();
  });

  it("renders the search input placeholder", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const inputElement = screen.getByPlaceholderText("Search on any topic...");
    expect(inputElement).toBeInTheDocument();
  });

  it("renders Browse by topics section", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const browseTitleElement = screen.getByText("Browse by topics");
    expect(browseTitleElement).toBeInTheDocument();
  });

  it("renders Frequently Asked Questions section", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const faqTitleElement = screen.getByText("Frequently Asked Questions");
    expect(faqTitleElement).toBeInTheDocument();
  });

  it("renders Contact Us button", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const contactButton = screen.getByText("Contact Us");
    expect(contactButton).toBeInTheDocument();
  });

  // Screen reader accessibility tests
  it("announces the Help Center section for screen readers", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const helpCenterSection = screen.getByLabelText("Help Center");
    expect(helpCenterSection).toBeInTheDocument();
  });

  it("announces the main heading for screen readers", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const mainHeading = screen.getByRole("heading", {
      level: 1,
      name: "How can we help You?",
    });
    expect(mainHeading).toBeInTheDocument();
  });

  it("announces the Browse by topics section for screen readers", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const browseTopicsSection = screen.getByLabelText("Browse by topics");
    expect(browseTopicsSection).toBeInTheDocument();
  });

  it("announces the Frequently Asked Questions section for screen readers", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const faqSection = screen.getByRole("heading", {
      level: 1,
      name: "Frequently Asked Questions",
    });
    expect(faqSection).toBeInTheDocument();
  });

  it("announces the search input for screen readers", () => {
    expect.hasAssertions();
    render(<HelpCenterPage />);
    const searchInput = screen.getByRole("textbox", {
      name: "Search on any topic",
    });
    expect(searchInput).toBeInTheDocument();
  });

  // Button test
  it("triggers an alert when the Contact Us button is clicked", () => {
    expect.hasAssertions();
    // Mock the window.alert function
    const alertMock = vi.fn();
    window.alert = alertMock;

    // Render the component
    render(<HelpCenterPage />);
    const contactButton = screen.getByText("Contact Us");
    fireEvent.click(contactButton);
    expect(alertMock).toHaveBeenCalledWith("Contact Button Click Test");
  });

  // Responsive test with snapshot
  it("renders correctly at different screen sizes", () => {
    expect.hasAssertions();
    const { container } = render(<HelpCenterPage />);

    // Set screen size to small (mobile)
    global.innerWidth = 320;
    global.dispatchEvent(new Event("resize"));
    expect(container).toMatchSnapshot("mobile");

    // Set screen size to medium (tablet)
    global.innerWidth = 768;
    global.dispatchEvent(new Event("resize"));
    expect(container).toMatchSnapshot("tablet");

    // Set screen size to large (desktop)
    global.innerWidth = 1024;
    global.dispatchEvent(new Event("resize"));
    expect(container).toMatchSnapshot("desktop");
  });
});
