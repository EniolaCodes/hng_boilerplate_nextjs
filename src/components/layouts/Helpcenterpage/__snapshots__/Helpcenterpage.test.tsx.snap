// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`helpCenterPage > renders correctly at different screen sizes > desktop 1`] = `
<div>
  <div
    class="w-full bg-background"
  >
    <div
      class="w-full bg-primary/10"
    >
      <div
        class="mx-auto w-full"
        style="max-width: 1349px;"
      >
        <section
          aria-labelledby="help-center-heading"
          class="flex w-full flex-col items-center justify-center gap-4 px-6 py-[24px] text-center md:px-0 md:py-24"
        >
          <span
            class="text-xl font-medium text-neutral-600"
            id="help-center-heading"
          >
            Help Center
          </span>
          <div
            class="flex h-48 flex-col items-center justify-center gap-5 self-stretch"
          >
            <h1
              aria-level="1"
              class="text-3xl font-bold text-neutral-950 md:text-6xl"
              role="heading"
            >
              How can we help You?
            </h1>
            <p
              class="text-center text-base font-normal text-neutral-600 md:text-lg"
            >
              Find advice and answers from our support team
            </p>
            <div
              class="flex h-fit w-full items-center justify-start overflow-hidden rounded-xl border border-slate-300 bg-white px-2 py-[2px] text-xs font-normal leading-none text-neutral-600 focus-visible:ring-0 focus-visible:ring-offset-0 md:w-[466px] md:rounded-3xl"
            >
              <span
                class="flex h-8 w-8 items-center justify-center p-1"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 20L16.2223 16.2156M18.3158 11.1579C18.3158 13.0563 17.5617 14.8769 16.2193 16.2193C14.8769 17.5617 13.0563 18.3158 11.1579 18.3158C9.2595 18.3158 7.43886 17.5617 6.0965 16.2193C4.75413 14.8769 4 13.0563 4 11.1579C4 9.2595 4.75413 7.43886 6.0965 6.0965C7.43886 4.75413 9.2595 4 11.1579 4C13.0563 4 14.8769 4.75413 16.2193 6.0965C17.5617 7.43886 18.3158 9.2595 18.3158 11.1579V11.1579Z"
                    stroke="#525252"
                    stroke-linecap="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
              <input
                aria-label="Search on any topic"
                class="flex h-10 rounded-md border border-input text-sm ring-[#CBD5E1] file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:ring-[#f97316] disabled:cursor-not-allowed disabled:opacity-50 w-full border-none bg-transparent px-1 py-2 focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-offset-0"
                placeholder="Search on any topic..."
              />
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      class="mx-auto w-full"
      style="max-width: 1349px;"
    >
      <section
        aria-labelledby="browse-topics-heading"
        class="flex flex-col items-center justify-start gap-7 px-6 py-24 md:px-24"
      >
        <span
          class="w-96 text-center text-3xl font-bold text-orange-500"
          id="browse-topics-heading"
        >
          Browse by topics
        </span>
        <div
          class="px grid w-full grid-cols-1 gap-x-[20px] gap-y-[12px] md:grid-cols-2 lg:grid-cols-3"
        >
          <div
            aria-labelledby="faq-heading-0"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-0"
              role="heading"
            >
              What is EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-1"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-1"
              role="heading"
            >
              How does EcoClean work?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-2"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-2"
              role="heading"
            >
              What are the key features of EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsum repellendus doloribus deserunt totam doloremque, quia quaerat veniam harum.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-3"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-3"
              role="heading"
            >
              Who can benefit from using EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-4"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-4"
              role="heading"
            >
              What are the system requirements for EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-5"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-5"
              role="heading"
            >
              How do I use EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-6"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-6"
              role="heading"
            >
              How do I store EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-7"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-7"
              role="heading"
            >
              How much does EcoClean cost?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-8"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-8"
              role="heading"
            >
              Are there any discounts available?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
        </div>
      </section>
      <section
        class="flex w-full flex-col items-center justify-center gap-20 px-6 pb-24 pt-12 md:px-24"
      >
        <div
          class="grid w-full grid-cols-1 justify-items-stretch gap-y-8 xl:grid-cols-2"
        >
          <div
            class="flex flex-col gap-3"
          >
            <h1
              aria-level="1"
              class="self-stretch text-4xl font-semibold text-neutral-600"
              role="heading"
            >
              Frequently Asked Questions
            </h1>
            <p
              class="self-stretch text-2xl font-normal text-neutral-600"
            >
              We couldnâ€™t answer your question?
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-outline text-outline-foreground shadow-sm border border-border hover:bg-outline-hover rounded-md px-3 text-xs h-8 w-28"
              role="button"
            >
              Contact us
            </button>
          </div>
          <div
            class="flex w-full"
          >
            <div
              class="flex w-full items-center justify-center bg-white md:ml-auto 2xl:w-[500px]"
            >
              Faq Accordions
            </div>
          </div>
        </div>
        <div
          class="flex w-full items-center justify-center text-center"
        >
          <div
            class="inline-flex flex-col items-center justify-center gap-4 p-6 px-0 md:px-6"
          >
            <h1
              class="text-3xl font-bold text-orange-500"
            >
              Didnâ€™t find an answer?
            </h1>
            <p
              class="text-lg font-normal text-neutral-600"
            >
              Contact us for more inquiries and information about our services.
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow h-10 rounded-md px-8"
              role="button"
            >
              Contact Us
            </button>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`helpCenterPage > renders correctly at different screen sizes > mobile 1`] = `
<div>
  <div
    class="w-full bg-background"
  >
    <div
      class="w-full bg-primary/10"
    >
      <div
        class="mx-auto w-full"
        style="max-width: 1349px;"
      >
        <section
          aria-labelledby="help-center-heading"
          class="flex w-full flex-col items-center justify-center gap-4 px-6 py-[24px] text-center md:px-0 md:py-24"
        >
          <span
            class="text-xl font-medium text-neutral-600"
            id="help-center-heading"
          >
            Help Center
          </span>
          <div
            class="flex h-48 flex-col items-center justify-center gap-5 self-stretch"
          >
            <h1
              aria-level="1"
              class="text-3xl font-bold text-neutral-950 md:text-6xl"
              role="heading"
            >
              How can we help You?
            </h1>
            <p
              class="text-center text-base font-normal text-neutral-600 md:text-lg"
            >
              Find advice and answers from our support team
            </p>
            <div
              class="flex h-fit w-full items-center justify-start overflow-hidden rounded-xl border border-slate-300 bg-white px-2 py-[2px] text-xs font-normal leading-none text-neutral-600 focus-visible:ring-0 focus-visible:ring-offset-0 md:w-[466px] md:rounded-3xl"
            >
              <span
                class="flex h-8 w-8 items-center justify-center p-1"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 20L16.2223 16.2156M18.3158 11.1579C18.3158 13.0563 17.5617 14.8769 16.2193 16.2193C14.8769 17.5617 13.0563 18.3158 11.1579 18.3158C9.2595 18.3158 7.43886 17.5617 6.0965 16.2193C4.75413 14.8769 4 13.0563 4 11.1579C4 9.2595 4.75413 7.43886 6.0965 6.0965C7.43886 4.75413 9.2595 4 11.1579 4C13.0563 4 14.8769 4.75413 16.2193 6.0965C17.5617 7.43886 18.3158 9.2595 18.3158 11.1579V11.1579Z"
                    stroke="#525252"
                    stroke-linecap="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
              <input
                aria-label="Search on any topic"
                class="flex h-10 rounded-md border border-input text-sm ring-[#CBD5E1] file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:ring-[#f97316] disabled:cursor-not-allowed disabled:opacity-50 w-full border-none bg-transparent px-1 py-2 focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-offset-0"
                placeholder="Search on any topic..."
              />
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      class="mx-auto w-full"
      style="max-width: 1349px;"
    >
      <section
        aria-labelledby="browse-topics-heading"
        class="flex flex-col items-center justify-start gap-7 px-6 py-24 md:px-24"
      >
        <span
          class="w-96 text-center text-3xl font-bold text-orange-500"
          id="browse-topics-heading"
        >
          Browse by topics
        </span>
        <div
          class="px grid w-full grid-cols-1 gap-x-[20px] gap-y-[12px] md:grid-cols-2 lg:grid-cols-3"
        >
          <div
            aria-labelledby="faq-heading-0"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-0"
              role="heading"
            >
              What is EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-1"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-1"
              role="heading"
            >
              How does EcoClean work?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-2"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-2"
              role="heading"
            >
              What are the key features of EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsum repellendus doloribus deserunt totam doloremque, quia quaerat veniam harum.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-3"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-3"
              role="heading"
            >
              Who can benefit from using EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-4"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-4"
              role="heading"
            >
              What are the system requirements for EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-5"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-5"
              role="heading"
            >
              How do I use EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-6"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-6"
              role="heading"
            >
              How do I store EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-7"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-7"
              role="heading"
            >
              How much does EcoClean cost?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-8"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-8"
              role="heading"
            >
              Are there any discounts available?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
        </div>
      </section>
      <section
        class="flex w-full flex-col items-center justify-center gap-20 px-6 pb-24 pt-12 md:px-24"
      >
        <div
          class="grid w-full grid-cols-1 justify-items-stretch gap-y-8 xl:grid-cols-2"
        >
          <div
            class="flex flex-col gap-3"
          >
            <h1
              aria-level="1"
              class="self-stretch text-4xl font-semibold text-neutral-600"
              role="heading"
            >
              Frequently Asked Questions
            </h1>
            <p
              class="self-stretch text-2xl font-normal text-neutral-600"
            >
              We couldnâ€™t answer your question?
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-outline text-outline-foreground shadow-sm border border-border hover:bg-outline-hover rounded-md px-3 text-xs h-8 w-28"
              role="button"
            >
              Contact us
            </button>
          </div>
          <div
            class="flex w-full"
          >
            <div
              class="flex w-full items-center justify-center bg-white md:ml-auto 2xl:w-[500px]"
            >
              Faq Accordions
            </div>
          </div>
        </div>
        <div
          class="flex w-full items-center justify-center text-center"
        >
          <div
            class="inline-flex flex-col items-center justify-center gap-4 p-6 px-0 md:px-6"
          >
            <h1
              class="text-3xl font-bold text-orange-500"
            >
              Didnâ€™t find an answer?
            </h1>
            <p
              class="text-lg font-normal text-neutral-600"
            >
              Contact us for more inquiries and information about our services.
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow h-10 rounded-md px-8"
              role="button"
            >
              Contact Us
            </button>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;

exports[`helpCenterPage > renders correctly at different screen sizes > tablet 1`] = `
<div>
  <div
    class="w-full bg-background"
  >
    <div
      class="w-full bg-primary/10"
    >
      <div
        class="mx-auto w-full"
        style="max-width: 1349px;"
      >
        <section
          aria-labelledby="help-center-heading"
          class="flex w-full flex-col items-center justify-center gap-4 px-6 py-[24px] text-center md:px-0 md:py-24"
        >
          <span
            class="text-xl font-medium text-neutral-600"
            id="help-center-heading"
          >
            Help Center
          </span>
          <div
            class="flex h-48 flex-col items-center justify-center gap-5 self-stretch"
          >
            <h1
              aria-level="1"
              class="text-3xl font-bold text-neutral-950 md:text-6xl"
              role="heading"
            >
              How can we help You?
            </h1>
            <p
              class="text-center text-base font-normal text-neutral-600 md:text-lg"
            >
              Find advice and answers from our support team
            </p>
            <div
              class="flex h-fit w-full items-center justify-start overflow-hidden rounded-xl border border-slate-300 bg-white px-2 py-[2px] text-xs font-normal leading-none text-neutral-600 focus-visible:ring-0 focus-visible:ring-offset-0 md:w-[466px] md:rounded-3xl"
            >
              <span
                class="flex h-8 w-8 items-center justify-center p-1"
              >
                <svg
                  aria-hidden="true"
                  fill="none"
                  height="24"
                  viewBox="0 0 24 24"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M20 20L16.2223 16.2156M18.3158 11.1579C18.3158 13.0563 17.5617 14.8769 16.2193 16.2193C14.8769 17.5617 13.0563 18.3158 11.1579 18.3158C9.2595 18.3158 7.43886 17.5617 6.0965 16.2193C4.75413 14.8769 4 13.0563 4 11.1579C4 9.2595 4.75413 7.43886 6.0965 6.0965C7.43886 4.75413 9.2595 4 11.1579 4C13.0563 4 14.8769 4.75413 16.2193 6.0965C17.5617 7.43886 18.3158 9.2595 18.3158 11.1579V11.1579Z"
                    stroke="#525252"
                    stroke-linecap="round"
                    stroke-width="1.5"
                  />
                </svg>
              </span>
              <input
                aria-label="Search on any topic"
                class="flex h-10 rounded-md border border-input text-sm ring-[#CBD5E1] file:border-0 file:bg-transparent file:text-sm file:font-medium placeholder:text-muted-foreground focus-visible:ring-[#f97316] disabled:cursor-not-allowed disabled:opacity-50 w-full border-none bg-transparent px-1 py-2 focus-visible:outline-none focus-visible:ring-0 focus-visible:ring-offset-0"
                placeholder="Search on any topic..."
              />
            </div>
          </div>
        </section>
      </div>
    </div>
    <div
      class="mx-auto w-full"
      style="max-width: 1349px;"
    >
      <section
        aria-labelledby="browse-topics-heading"
        class="flex flex-col items-center justify-start gap-7 px-6 py-24 md:px-24"
      >
        <span
          class="w-96 text-center text-3xl font-bold text-orange-500"
          id="browse-topics-heading"
        >
          Browse by topics
        </span>
        <div
          class="px grid w-full grid-cols-1 gap-x-[20px] gap-y-[12px] md:grid-cols-2 lg:grid-cols-3"
        >
          <div
            aria-labelledby="faq-heading-0"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-0"
              role="heading"
            >
              What is EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-1"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-1"
              role="heading"
            >
              How does EcoClean work?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-2"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-2"
              role="heading"
            >
              What are the key features of EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ipsum repellendus doloribus deserunt totam doloremque, quia quaerat veniam harum.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-3"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-3"
              role="heading"
            >
              Who can benefit from using EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-4"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-4"
              role="heading"
            >
              What are the system requirements for EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-5"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-5"
              role="heading"
            >
              How do I use EcoClean?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-6"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-6"
              role="heading"
            >
              How do I store EcoClean?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-7"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-7"
              role="heading"
            >
              How much does EcoClean cost?
            </div>
            <div
              class="hidden"
            >
              Lorem ipsum dolor sit amet consectetur adipisicing elit.
            </div>
          </div>
          <div
            aria-labelledby="faq-heading-8"
            class="flex w-full flex-col bg-white px-4 py-8"
            role="region"
          >
            <div
              aria-level="2"
              class="border-b-0"
              id="faq-heading-8"
              role="heading"
            >
              Are there any discounts available?
            </div>
            <div
              class="hidden"
            >
              Yes. It adheres to the WAI-ARIA design pattern.
            </div>
          </div>
        </div>
      </section>
      <section
        class="flex w-full flex-col items-center justify-center gap-20 px-6 pb-24 pt-12 md:px-24"
      >
        <div
          class="grid w-full grid-cols-1 justify-items-stretch gap-y-8 xl:grid-cols-2"
        >
          <div
            class="flex flex-col gap-3"
          >
            <h1
              aria-level="1"
              class="self-stretch text-4xl font-semibold text-neutral-600"
              role="heading"
            >
              Frequently Asked Questions
            </h1>
            <p
              class="self-stretch text-2xl font-normal text-neutral-600"
            >
              We couldnâ€™t answer your question?
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-outline text-outline-foreground shadow-sm border border-border hover:bg-outline-hover rounded-md px-3 text-xs h-8 w-28"
              role="button"
            >
              Contact us
            </button>
          </div>
          <div
            class="flex w-full"
          >
            <div
              class="flex w-full items-center justify-center bg-white md:ml-auto 2xl:w-[500px]"
            >
              Faq Accordions
            </div>
          </div>
        </div>
        <div
          class="flex w-full items-center justify-center text-center"
        >
          <div
            class="inline-flex flex-col items-center justify-center gap-4 p-6 px-0 md:px-6"
          >
            <h1
              class="text-3xl font-bold text-orange-500"
            >
              Didnâ€™t find an answer?
            </h1>
            <p
              class="text-lg font-normal text-neutral-600"
            >
              Contact us for more inquiries and information about our services.
            </p>
            <button
              class="inline-flex items-center justify-center gap-2 whitespace-nowrap text-sm font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground shadow h-10 rounded-md px-8"
              role="button"
            >
              Contact Us
            </button>
          </div>
        </div>
      </section>
    </div>
  </div>
</div>
`;
